fix!: update token endpoints to non-deprecated endpoints

ddelgrosso1 opened the pull request:
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/node-gtoken/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #388  🦕

ddelgrosso1 committed:
fix: update token endpoints to non-deprecated endpoints

ddelgrosso1 commented:
Both of the endpoints changed in this PR are marked as `deprecated` / `extremely deprecated` per documentation. Please feel free to reach out for more information. 

I also cleaned up some unused variables in tests that were giving linter warnings.

sai-sunder-s commented on src/index.ts:
```diff
@@ -21,9 +21,9 @@ const readFile = fs.readFile
+const GOOGLE_TOKEN_URL = 'https://oauth2.googleapis.com/token';
 const GOOGLE_REVOKE_TOKEN_URL =
-  'https://accounts.google.com/o/oauth2/revoke?token=';
+  'https://oauth2.googleapis.com/token/revoke?token=';
```
where did you get this revoke url from?

ddelgrosso1 replied:
In the documentation it said that the revoke old url (accounts.google.com) was `extremely deprecated`, so I browsed through the proto from the yaqs issue and saw revoke was in there.

sai-sunder-s replied:
Got it. As long as it works, it should be ok.

ddelgrosso1 committed:
fix revoke url

ddelgrosso1 commented on src/index.ts:
```diff
@@ -21,9 +21,8 @@ const readFile = fs.readFile
-const GOOGLE_TOKEN_URL = 'https://www.googleapis.com/oauth2/v4/token';
-const GOOGLE_REVOKE_TOKEN_URL =
-  'https://accounts.google.com/o/oauth2/revoke?token=';
+const GOOGLE_TOKEN_URL = 'https://oauth2.googleapis.com/token';
```
https://github.com/googleapis/google-api-php-client/blob/7a95ed29e4b6c6859d2d22300c5455a92e2622ad/src/Client.php#L56

ddelgrosso1 committed:
Merge branch 'main' into update-endpoints

ddelgrosso1 commented on src/index.ts:
```diff
@@ -21,9 +21,8 @@ const readFile = fs.readFile
-const GOOGLE_REVOKE_TOKEN_URL =
-  'https://accounts.google.com/o/oauth2/revoke?token=';
+const GOOGLE_TOKEN_URL = 'https://oauth2.googleapis.com/token';
+const GOOGLE_REVOKE_TOKEN_URL = 'https://oauth2.googleapis.com/revoke?token=';
```
https://github.com/googleapis/google-auth-library-ruby/blob/2574faf1474711e581cdfde508387630ea3c4b8f/lib/googleauth/user_refresh.rb#L36

d-goog committed:
Merge branch 'main' of github.com:googleapis/node-gtoken into update-endpoints

d-goog committed:
chore: remove log

d-goog committed:
chore: lint fix

d-goog committed:
fix: reserved

One of the layers of this build system throws with 'reserved' keyword, while typescript itself doesn't

d-goog committed:
style: lint

d-goog commented:
Merged and will further test this in auth downstream via the release candidate of this library.
